virtualenv.py

```diff
@@ -1376,8 +1379,9 @@ def install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages, clear, sy
         # the value:
         py_executable = '"%s"' % py_executable
     # NOTE: keep this check as one line, cmd.exe doesn't cope with line breaks
-    cmd = [py_executable, '-c', 'import sys;out=sys.stdout;'
-        'getattr(out, "buffer", out).write(sys.prefix.encode("utf-8"))']
+    cmd = [py_executable, '-c', 'import sys;out=sys.stdout;prefix=sys.prefix;'
+        'prefix=prefix if sys.version_info[0]==2 else prefix.encode(sys.getfilesystemencoding());'
+        'getattr(out, "buffer", out).write(prefix)']
```

Comment:
> why are you converting the protocol to use "string if Python 2 otherwise bytes" rather than a clean always-bytes protocol?

Actually my codes implement an "always-bytes protocol". On Python 2, ```sys.prefix``` is a bytes-like object, so I print it as-is.

> What would happen with your code if sys.prefix contained characters that weren't encodable in sys.getfilesystemencoding()?

Maybe sys.stdout.encoding is a better choice here.